pkg/bootstrap/config.go

```diff
@@ -278,9 +280,20 @@ func getServiceCluster(metadata *model.BootstrapNodeMetadata) string {
 		} else if metadata.Namespace != "" {
 			serviceCluster = "istio-proxy." + metadata.Namespace
 		}
+		return serviceCluster
 	}
-
-	return serviceCluster
+	tmpl, err := template.New("serviceCluster").Parse(serviceCluster)
+	if err != nil {
+		log.Infof("could not parse template: %v", err)
+		return serviceCluster
+	}
+	buf := &bytes.Buffer{}
+	err = tmpl.Execute(buf, metadata)
+	if err != nil {
+		log.Warnf("could not execute template: %v", err)
+		return serviceCluster
```

Comment:
Returning the template does not sound right.

Reply:
fixed.